core/src/main/java/com/linecorp/armeria/common/RequestHeaders.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Consumer;
+
+/**
+ * Immutable HTTP/2 headers for an {@link HttpRequest}.
+ *
+ * @see ResponseHeaders
+ */
+public interface RequestHeaders extends HttpHeaders, RequestHeaderGetters {
+
+    /**
+     * Creates a new empty builder.
+     */
+    static RequestHeadersBuilder builder() {
+        return new DefaultRequestHeadersBuilder();
+    }
+
+    /**
+     * Creates a new builder with the specified {@link HttpMethod} and {@code path}.
```

Comment:
I think there is inconsistency in javadoc among the classes e.g., `{@code ":path"}`, `{@code path}`. Would personally lean towards `{@code :path}` or `{@code path}` since quotes seems redundant when using at code.

Reply:
`":path"` refers to the HTTP/2 `:path` pseudo header and `path` refers to the parameter `path`.

Reply:
So, we wrap header names with double quotes and do not for parameter names, so I think `":path"` and `path` are consistent.

Reply:
Makes sense - I got confused since they're both together in this PR, but when reading documentation as normal it wouldn't be an issue :)